tests/plugins/stream.rs

```diff
@@ -190,3 +190,14 @@ fn generate_sequence() {
 
     assert_eq!(actual.out, "[0,2,4,6,8,10]");
 }
+
+#[test]
+#[ntest::timeout(6000)]
+fn echo_interactivity_on_slow_pipelines() {
+    let actual = nu_with_plugins!(
+        cwd: "tests/fixtures/formats",
+        plugin: ("nu_plugin_example"),
+        r#"[1] | each { |n| sleep 10sec; $n } | prepend 0 | example echo | first"#
```

Comment:
Huhm shouldn't this fail with 10 s of sleep and 6 s of timeout?

Reply:
I think the `prepend` is supposed to prevent the pipeline from sleeping for the first value.

Reply:
The desired (and current) behavior is that `[1] | each { |n| sleep 10sec; $n } | prepend 0 | example echo | first` returns immediately. If the plugin pipeline isn't streaming then this does take >10s, as it needs to wait for the 1. When working though, 0 is available immediately and the first completes the pipeline.

Reply:
Ah right, this `each` should indeed act lazy and if `first` wouldn't consume parsimonous (and the `example echo` plugin command wouldn't be streaming) then the test would sleep into the timeout and fail. Clever

Reply:
I can't take credit, it was a suggestion from @devyn 😅

Reply:
I've pushed up a comment to describe the behavior 👍